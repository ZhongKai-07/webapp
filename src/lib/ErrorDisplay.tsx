import React, { PropsWithChildren } from 'react';

import MapIcon from '@/lib/MapIcon';

interface ErrorDisplayProps extends PropsWithChildren {
  message: string;
  isMapError?: boolean;
}
const ErrorDisplay: React.FC<ErrorDisplayProps> = ({ message, isMapError }) => {
  return (
    <div role='alert' className='px-4'>
      <div className='bg-red-500 text-white font-bold rounded-t px-4 py-2 '>
        Warning
      </div>
      <div
        className='border border-t-0 border-red-400 rounded-b bg-red-100 px-4 py-3 text-red-700'
        style={{ height: '75%' }}
      >
        {isMapError && (
          <div
            className='grid grid-cols-1 grid-rows-2 gap-12'
            style={{ height: '100%' }}
          >
            <div className='my-4'>
              <div className='flex justify-center items-center'>
                <MapIcon className='text-red-700' testId='map-icon' />
              </div>
            </div>
            <div className='row-start-2 my-4'>
              <p>{message}</p>
            </div>
          </div>
        )}
        {!isMapError && <p>{message}</p>}
      </div>
    </div>
  );
};

export default ErrorDisplay;
