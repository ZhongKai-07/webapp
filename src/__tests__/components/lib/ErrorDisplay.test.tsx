import { render } from '@testing-library/react';
import React from 'react';

import ErrorDisplay from '@/lib/ErrorDisplay';

describe('ErrorDisplay', () => {
  it('should render correctly', () => {
    const message = 'This is an error message';
    const isMapError = true;

    const { container } = render(
      <ErrorDisplay message={message} isMapError={isMapError} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render a warning message', () => {
    const message = 'This is an error message';
    const isMapError = true;

    const { getByText } = render(
      <ErrorDisplay message={message} isMapError={isMapError} />
    );

    const warningMessage = getByText('Warning');
    expect(warningMessage).toBeInTheDocument();
  });

  it('should render the error message', () => {
    const message = 'This is an error message';
    const isMapError = true;

    const { getByText } = render(
      <ErrorDisplay message={message} isMapError={isMapError} />
    );

    const errorMessage = getByText(message);
    expect(errorMessage).toBeInTheDocument();
  });

  it('should render the MapIcon when isMapError is true', () => {
    const message = 'This is an error message';
    const isMapError = true;

    const { getByTestId } = render(
      <ErrorDisplay message={message} isMapError={isMapError} />
    );

    const mapIcon = getByTestId('map-icon');
    expect(mapIcon).toBeInTheDocument();
  });

  it('should not render the MapIcon when isMapError is false', () => {
    const message = 'This is an error message';
    const isMapError = false;

    const { queryByTestId } = render(
      <ErrorDisplay message={message} isMapError={isMapError} />
    );

    const mapIcon = queryByTestId('map-icon');
    expect(mapIcon).toBeNull();
  });
});
