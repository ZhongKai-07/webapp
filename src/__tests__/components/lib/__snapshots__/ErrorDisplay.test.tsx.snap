// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay should render correctly 1`] = `
<div>
  <div
    class="px-4"
    role="alert"
  >
    <div
      class="bg-red-500 text-white font-bold rounded-t px-4 py-2 "
    >
      Warning
    </div>
    <div
      class="border border-t-0 border-red-400 rounded-b bg-red-100 px-4 py-3 text-red-700"
      style="height: 75%;"
    >
      <div
        class="grid grid-cols-1 grid-rows-2 gap-12"
        style="height: 100%;"
      >
        <div
          class="my-4"
        >
          <div
            class="flex justify-center items-center"
          >
            <svg
              aria-hidden="true"
              class="h-10 w-10 text-red-700"
              data-testid="map-icon"
              fill="currentColor"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.143 37.881c4.178-6.076 9.957-15.809 9.957-23.164a14.718 14.718 0 0 0-29.435 0c0 7.357 5.783 17.094 9.961 23.168C6.3 38.391 0 40.074 0 42.858 0 46.235 9.248 48 18.383 48s18.384-1.766 18.384-5.143c0-2.782-6.293-4.47-13.624-4.976zM5.121 14.717a13.264 13.264 0 0 1 26.527 0c0 9.492-10.665 23.852-13.264 27.209-2.598-3.357-13.263-17.713-13.263-27.209zm13.262 31.829c-10.334 0-16.929-2.184-16.929-3.688 0-1.309 5.03-3.123 13.155-3.57a91.912 91.912 0 0 0 3.209 4.273.729.729 0 0 0 1.129 0 93.58 93.58 0 0 0 3.213-4.277c8.125.447 13.151 2.266 13.151 3.574.002 1.504-6.593 3.688-16.928 3.688z"
                />
                <path
                  d="M26.383 14.848a8 8 0 1 0-8 8 8.009 8.009 0 0 0 8-8zm-14.545 0a6.546 6.546 0 1 1 6.545 6.545 6.555 6.555 0 0 1-6.545-6.545z"
                />
              </g>
            </svg>
          </div>
        </div>
        <div
          class="row-start-2 my-4"
        >
          <p>
            This is an error message
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
